### ISSUE TITLE
ecmaVersion 5 ignored

### ISSUE THREAD

**Issue by mjomble** [babel/babel-eslint#431]
If I have a file with ES2015 code like this:
```javascript
var [foo, bar] = []
```
then ESLint with default configuration gives a "Parsing error: Unexpected token [" error because the default ecmaVersion is 5.

If I use babel-eslint as the parser, I won't get an error even if I change the configuration to

```javascript
module.exports = {
  parser: 'babel-eslint',
  parserOptions: { sourceType: 'script', ecmaVersion: 5 }
}
```

Is this expected behavior or a bug?
I'm guessing maybe `ecmaVersion` isn't even a valid option for this parser.
If so, perhaps this could be made more explicitly clear in the README.

I have a project where some of the code uses class properties, Flow annotations, etc (so I need babel-eslint) while some of the code must be runnable on older browsers and therefore can't use any ES2015+ features (so I need the default parser?).
Do you have recommendations for configuring ESLint on such a project so that using ES2015+ features in this legacy code would also trigger linter errors?

**Comment by hzoo** [comment]
It's ignored because we assume if you are running with babel-eslint you are writing and using ES6/Babel. I think you would have to figure out running with 2 different eslint configs/options based on the folder or something like that

**Comment by hzoo** [comment]
There's no way to prevent syntax from being written via this plugin atm http://eslint.org/docs/rules/no-restricted-syntax https://github.com/amilajack/eslint-plugin-compat maybe

**Comment by hzoo** [comment]
can follow https://github.com/babel/babel-eslint/issues/431 actually as a dup
